src/rebar_dir.erl

```diff
@@ -133,16 +133,22 @@ home_dir() ->
     {ok, [[Home]]} = init:get_argument(home),
     Home.
 
+basedir(Type) ->
+    filename:basedir(Type, "rebar3", #{os => linux}).
```

Comment:
Oh, hm, if we are going to do this we probably shouldn't hardcode `linux`. It may change where stuff goes in windows and osx but probably better to be right than continue being wrong? But not sure... 

@ferd thoughts?

Reply:
Replace my "may" with "will". https://erlang.org/doc/man/filename.html#basedir-3

Reply:
I am on macOS and I prefer my development tooling to use XDG rather than macOS practices ;)

Reply:
This may unfortunately blow out things in the global rebar.config file and [existing hex auth](https://github.com/erlang/rebar3/blob/master/src/rebar_hex_repos.erl#L173-L175) and be considered backwards incompatible in very surprising ways.

If we're to support the new stuff, it should ideally come with supporting both at once with a warning or information message asking the user to move content, which they may hate if they have to use various rebar3 versions for various OTP support (or when building with mix, which bundles older ones).

Forcing the linux path likely keeps things compatible with the current mechanism on all operating systems though, since it's what we forced everywhere (would have to double check windows)

Reply:
Yea. I suppose we can wait for rebar4.

Reply:
@ferd in what way it can blow things? It would only change behaviour if user have set `XDG_CONFIG_HOME`.

Reply:
Using it with the linux path everywhere should remain the same although there's a check needed for windows because I don't know if they're fully equivalent. Following the basedir fully would break stuff.

Reply:
@ferd what check for windows are you referring to?

I think this PR is good to merge except for a question I have on line 151.

Reply:
I think we're fine if we keep forcing the linux format. I just double-checked, the thing that worried me is that there could be a conversion issue between what was considered the home_dir() in windows as of today vs. how the Linux path would translate.

I did a check on my old win10 machine and it seems to be identical and good to go.